org-roam-buffer.el

```diff
@@ -0,0 +1,266 @@
+;; This library provides the org-roam-buffer functionality for org-roam
+;;; Code:
+;;;; Library Requires
+(require 'cl-lib)
+(require 'dash)
+(require 's)
+
+(defvar org-roam-directory)
+(defvar org-link-frame-setup)
+(defvar org-return-follows-link)
+(defvar org-roam-backlinks-mode)
+(defvar org-roam-last-window)
+(declare-function org-roam-db--ensure-built   "org-roam-db")
+(declare-function org-roam--extract-ref       "org-roam")
+(declare-function org-roam--get-title-or-slug "org-roam")
+(declare-function org-roam--get-backlinks     "org-roam")
+(declare-function org-roam-backlinks-mode     "org-roam")
+
+(defcustom org-roam-buffer-position 'right
+  "Position of `org-roam' buffer.
+Valid values are
+ * left,
+ * right,
+ * top,
+ * bottom."
+  :type '(choice (const left)
+                 (const right)
+                 (const top)
+                 (const bottom))
+  :group 'org-roam)
+
+(defcustom org-roam-buffer-width 0.33
+  "Width of `org-roam' buffer.
+Has an effect if and only if `org-roam-buffer-position' is `left' or `right'."
+  :type 'number
+  :group 'org-roam)
+
+(defcustom org-roam-buffer-height 0.27
+  "Height of `org-roam' buffer.
+Has an effect if and only if `org-roam-buffer-position' is `top' or `bottom'."
+  :type 'number
+  :group 'org-roam)
+
+
+(defcustom org-roam-buffer "*org-roam*"
+  "Org-roam buffer name."
+  :type 'string
+  :group 'org-roam)
+
+(defcustom org-roam-buffer-prepare-hook '(org-roam-buffer--insert-title
+                                          org-roam-buffer--insert-backlinks
+                                          org-roam-buffer--insert-citelinks)
+  "Hook run in the `org-roam-buffer' before it is displayed."
+  :type 'hook
+  :group 'org-roam)
+
+(defalias               'org-roam--current-buffer 'org-roam-buffer--current)
+(make-obsolete-variable 'org-roam--current-buffer 'org-roam-buffer--current "2020/04/06")
+(defvar org-roam-buffer--current nil
+  "Currently displayed file in `org-roam' buffer.")
+
+(defun org-roam-buffer--insert-title ()
+  "Insert the org-roam-buffer title."
+  (insert (propertize (org-roam--get-title-or-slug
+                       (buffer-file-name org-roam-buffer--current))
+                       'font-lock-face
+                       'org-document-title)))
+
+(defun org-roam-buffer--insert-citelinks ()
+  "Insert citation backlinks for the current buffer."
+  (if-let* ((roam-key (with-temp-buffer
+                        (insert-buffer-substring org-roam-buffer--current)
+                        (org-roam--extract-ref)))
+            (key-backlinks (org-roam--get-backlinks (s-chop-prefix "cite:" roam-key)))
+            (grouped-backlinks (--group-by (nth 0 it) key-backlinks)))
+      (progn
+        (insert (format "\n\n* %d Cite backlinks\n"
+                        (length key-backlinks)))
+        (dolist (group grouped-backlinks)
+          (let ((file-from (car group))
+                (bls (cdr group)))
+            (insert (format "** [[file:%s][%s]]\n"
```

Comment:
Maybe also use `(org-link-make-string link description)` here (and in backlinks function), as it will take care of proper escaping?

Reply:
I agree this is the right thing to do. Probably want to wrap `org-link-make-string' in a conditional alias so we can cater to pre 9.3 versions of Org as well. I'll look into this today.

Reply:
See: #412 This would allow us to take care of backward compatibility in a clean way.